### PR TITLE
feature(node_benchmarks): add result comparison

### PR DESCRIPTION
**Pull request by KnifeyMoloko**
This PR adds 2 new features for the recently introduced node benchmarks:
- comparing the current results against previous results for a given node type
- checking if the possible difference between current node benchmarks and the previous runs average is within a defined tolerance margin
- adding node benchmarks to some of our longevity runs

Email table with results example:

image

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/report_templates/results_longevity.html [hunk 1]
```diff
@@ -9,6 +9,44 @@ <h3>
...
+                    <tr>
+                        <td>{{ item|first }}</td>
+                        {% for name in node_benchmarks|dictsort|first|last|sort %}
+                        <td>
+                            <table class="'">
+                                {% for key, value in (item|last)[name].items() %}
+                                <tr>
+                                    <td style="font-weight: bold">{{ key }}</td>
+                                    {% if value is boolean() %}
+                                    {% if value is true() %}
```
Comment (fgelcer) [inline sdcm/report_templates/results_longevity.html:32]: nitpick: missing indentation

**Review by fgelcer — COMMENTED** [review]
do you have a print screen with the result as an example of it?
also, a single comment about `.html` file indentation, but other than that, LGTM

File: sdcm/report_templates/results_longevity.html (hunk 1 above)
Comment (KnifeyMoloko) [reply sdcm/report_templates/results_longevity.html]: Fixed

**Comment by KnifeyMoloko** [comment]
> do you have a print screen with the result as an example of it? also, a single comment about `.html` file indentation, but other than that, LGTM

@fgelcer I added an example screenshot in the PR description

**Comment by fgelcer** [comment]
> > do you have a print screen with the result as an example of it? also, a single comment about `.html` file indentation, but other than that, LGTM
> 
> @fgelcer I added an example screenshot in the PR description

i see it, and i suggest, to improve readability to limit these average values to have up to 2 decimal

**Comment by fgelcer** [comment]
WDYT?

**Comment by KnifeyMoloko** [comment]
> > > do you have a print screen with the result as an example of it? also, a single comment about `.html` file indentation, but other than that, LGTM
> > 
> > 
> > @fgelcer I added an example screenshot in the PR description
> 
> i see it, and i suggest, to improve readability to limit these average values to have up to 2 decimal

@fgelcer Good idea. I added the formatting to the `results_longevity.html` template.

**Review by fgelcer — APPROVED** [review]
LGTM.

it will be nice to see the updated result (with 2 decimal), but approved.

File: sdcm/report_templates/results_longevity.html [hunk 2]
```diff
@@ -9,6 +9,44 @@ <h3>
             <li>Instance type: {{ scylla_instance_type }}</li>
             <li>Number of scylladb nodes: {{ number_of_db_nodes }}</li>
 
+            {% if node_benchmarks is not eq("N/A") %}
```
Comment (enaydanov) [inline sdcm/report_templates/results_longevity.html:12]: Is it a correct comparison? As I can see `node_benchmarks` can be `None` or a dict.

Comment (KnifeyMoloko) [reply sdcm/report_templates/results_longevity.html:12]: I got confused by the `BaseEmailReporter` and it's `build_data_for_report()` method which replaces missing dict values with "N/A". Looks like `{% if node_benchmarks %}` should do it here.

**Review by enaydanov — APPROVED** [review]

**Review by rayakurl — APPROVED** [review]
Marking as 'approved'  since 
@fgelcer approved it, bit he's currently on a vacation

**Pull request merged**